micrometer-jmx/src/main/scala/com/avast/sst/micrometer/jmx/TypeScopeNameObjectNameFactory.scala

```diff
@@ -0,0 +1,51 @@
+package com.avast.sst.micrometer.jmx
+
+import java.util
+import java.util.regex.Pattern
+
+import com.codahale.metrics.jmx.{DefaultObjectNameFactory, ObjectNameFactory}
+import javax.management.ObjectName
+
+import scala.util.Try;
+
+/** This is custom [[com.codahale.metrics.jmx.ObjectNameFactory]] which uses "type-scope-name" hierarchy of resulting
+  * [[javax.management.ObjectName]] (levels 3-N are glued together).
+  */
+class TypeScopeNameObjectNameFactory(separator: String = ".") extends ObjectNameFactory {
```

Comment:
does this class need  to be public? can we just not return an instance of `ObjectNameFactory`?

Reply:
Not really, I can hide it from everyone. I just thought it could be useful to someone but you are right it's better to hide it.